File: mobile_app/src/models/device.ts
export class DeviceModel {
  constructor(public id: number,
              public name: string,
              public manufacturer: string,
              public model: string,
              public serial_number: string,
              public mac_address: string,
              public validic_id: number,
              public image_url: string,
              public peripheral_type: string,
              public paired: boolean,
              public last_reading: Date,
              public active: boolean) {
  }
  
  readingLabel() {
    switch (this.peripheral_type) {
      case ('thermometer'):
        return'&#8457';
        //break;
      case ('blood_pressure'):
        return 'mm Hg';
        //break;
      case ('heart_rate'):
        return 'bpm';
        //break;
      case ('glucose'):
        return'level';
        //break;
      case ('weight'):
        return 'lbs';
        //break;
      case ('pulse_oximeter'):
        return '%';
        //break;
      case ('fitness_tracker'):
        return 'level';
        //break;
    }
  }
}

// JDavis: todo - add device_id from HE app.

/*  JDavis: peripheral_types
    none: 0,
    thermometer: 1,
    blood_pressure: 2,
    heart_rate: 3,
    glucose: 4,
    weight: 5,
    pulse_oximeter: 6
    fitness_tracker:
*/

File: adherece_oracle/app/models/device.rb
# == Schema Information
#
# Table name: devices
#
#  id                   :integer          not null, primary key
#  name                 :string
#  vendor               :string
#  vendor_number        :string
#  cost                 :decimal(, )
#  created_at           :datetime         not null
#  updated_at           :datetime         not null
#  validic_id           :integer
#  manufacturer         :string
#  model                :string
#  pairing_instructions :text
#  instructions         :text
#  reading_instructions :text
#  image_url            :string
#  peripheral_type      :string
#  device_sfid          :string
#

class Device < ApplicationRecord
  
  has_many  :patient_devices, foreign_key: :device_sfid, primary_key: :device_sfid
  has_many  :patients, through: :patient_devices
  
  after_create :ensure_device_sfid
  
  def full_name
    "#{manufacturer} #{model}"
  end
  
  def self.peripheral_types
    ['blood_pressure', 'fitness_tracker', 'gateway', 'glucose', 'heart_rate', 'pulse_oximeter', 'thermometer', 'weight']
  end
  
  def ensure_device_sfid
    if Rails.env == 'development'
      self.update_attribute(:device_sfid, self.id) unless self.device_sfid.exists?
    end
  end
  
end

    # none: 0,
    # thermometer: 1,
    # blood_pressure: 2,
    # heart_rate: 3,
    # glucose: 4,
    # weight: 5,
    # pulse_oximeter: 6
    # fitness_tracker: 
